### PR TITLE
dev-881_stripe_disable_wallets

### PR DESCRIPTION
**Pull request by gurjittoptal**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
draft

#### What's this PR do?
disables applePay, googlePay or browserCard based on donation Page settiing.

#### Why are we doing this? How does it help us?
This is a feature thats been requested by some orgs testing the app.

#### How should this be manually tested?
googlePay disabled
https://billypenn-dev-881.revengine-review.org/donate 
googlePay not disabled
https://newsrevenuehub-dev-881.revengine-review.org/donate 

Similarly check the other conditions by editing payment provider for donation pages

#### Have automated unit tests been added?
no

#### How should this change be communicated to end users?
n/a

#### Are there any smells or added technical debt to note?
no

#### Has this been documented? If so, where?
no

#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-881

### CODE REVIEW

File: spa/src/components/paymentProviders/stripe/StripePaymentForm.js [hunk 1]
```diff
@@ -218,14 +218,36 @@ function StripePaymentForm({ loading, setLoading, offerPayFees }) {
   useEffect(() => {
     const rpIsNonProfit = page.revenue_program_is_nonprofit;
     const amnt = amountToCents(getTotalAmount(amount, payFee, frequency, rpIsNonProfit));
+
+    let disabledWallets = [];
+
+    if (page.elements) {
+      for (let ind in page.elements) {
+        let element = page.elements[ind];
+
```
Comment (benjaminEwhite) [inline spa/src/components/paymentProviders/stripe/StripePaymentForm.js:227]: I think something like this makes the idea of this a bit clearer and more succinct...

```javascript
    (page?.elements || [])
      .filter((elem) => elem.type === 'Dpayment')
      .forEach((paymentType) => {
        ....
      })

```

Comment (benjaminEwhite) [reply spa/src/components/paymentProviders/stripe/StripePaymentForm.js:227]: Also, is it ever the case that there's actually > 1 Dpayment element? If not, `.find` would make more sense here.

**Review by benjaminEwhite — APPROVED** [review]
Have a suggestion for what is IMO a way to code this where the intention is a little clearer, but that may be a personal preference.

File: spa/src/components/paymentProviders/stripe/StripePaymentForm.js (hunk 1 above)
Comment (gurjittoptal) [reply spa/src/components/paymentProviders/stripe/StripePaymentForm.js]: done.

File: spa/src/components/paymentProviders/stripe/StripePaymentForm.js [hunk 2]
```diff
@@ -218,14 +218,33 @@ function StripePaymentForm({ loading, setLoading, offerPayFees }) {
   useEffect(() => {
     const rpIsNonProfit = page.revenue_program_is_nonprofit;
     const amnt = amountToCents(getTotalAmount(amount, payFee, frequency, rpIsNonProfit));
+    let disabledWallets = [];
+
+    (page?.elements || [])
+      .filter((elem) => elem.type === 'DPayment')
+      .forEach((paymentType) => {
+        let stripeDisabled = paymentType.content.stripe;
```
Comment (benjaminEwhite) [inline spa/src/components/paymentProviders/stripe/StripePaymentForm.js:226]: believe this should be called `stripeEnabled`, no?

Comment (gurjittoptal) [reply spa/src/components/paymentProviders/stripe/StripePaymentForm.js]: yes, changed to enabledWalletsForPage

File: spa/src/components/paymentProviders/stripe/StripePaymentForm.test.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    const page = {
+      elements: [{ type: 'DPayment', content: { stripe: ['card', 'browser', 'apple', 'google'], offerPayFees: true } }]
+    };
+    result = await getDisabledWallets(page);
+  });
+
+  it('should return empty array', () => {
+    expect(result.length).toEqual(0);
+  });
+});
```
Comment (benjaminEwhite) [inline spa/src/components/paymentProviders/stripe/StripePaymentForm.test.js:59]:
```suggestion
import isEqual from 'lodash.isequal';

import { getDisabledWallets } from './StripePaymentForm';

describe('getDisabledWallets function', () => {
  it.each`
    pagePaymentTypes                  | expectedSet
    ${['browser', 'apple']}           | ${new Set(['googlePay'])}
    ${['browser', 'google']}          | ${new Set(['applePay'])}
    ${['apple', 'google']}            | ${new Set(['browserCard'])}
    ${[]}                             | ${new Set(['browserCard', 'applePay', 'googlePay'])}
    ${['browser', 'apple', 'google']} | ${new Set([])}
  `('returns list equivalent of $expectedSet when given $pagePaymentTypes', ({ pagePaymentTypes, expectedSet }) => {
    const page = {
      elements: [{ type: 'DPayment', content: { stripe: pagePaymentTypes, offerPayFees: true } }]
    };
    expect(isEqual(new Set(getDisabledWallets(page)), expectedSet)).toBe(true);
  });
});

```

**Review by benjaminEwhite — COMMENTED** [review]
Left a suggestion for simplifying/cleaning up the test. Aside from that I'm 👍

**Review by x110dc — APPROVED** [review]

**Pull request merged**